galaxy_ng/tests/integration/api/test_ui_paths.py

```diff
@@ -0,0 +1,684 @@
+        assert resp.status_code == 200
+
+        ds = resp.json()
+        validate_json(instance=ds, schema=schema_objectlist)
+
+        for distro in ds['data']:
+            validate_json(instance=distro, schema=schema_distro)
+
+        # check the endpoint for each distro by pulp id ...
+        distro_ids = [x['pulp_id'] for x in ds['data']]
+        for distro_id in distro_ids:
+            resp = uclient.get(f'_ui/v1/distributions/{distro_id}')
+            assert resp.status_code == 200
+            _ds = resp.json()
+            validate_json(instance=_ds, schema=schema_distro)
+            validate_json(instance=_ds['repository'], schema=schema_distro_repository)
+            assert _ds['pulp_id'] == distro_id
+
+
+# /api/automation-hub/_ui/v1/execution-environments/namespaces/
+# /api/automation-hub/_ui/v1/execution-environments/namespaces/{name}/
+
+
+# /api/automation-hub/_ui/v1/execution-environments/registries/
+@pytest.mark.standalone_only
+@pytest.mark.api_ui
+def test_api_ui_v1_execution_environments_registries(ansible_config):
+
+    cfg = ansible_config('ansible_partner')
+    with UIClient(config=cfg) as uclient:
+
+        # get the response
+        resp = uclient.get('_ui/v1/execution-environments/registries/')
+        assert resp.status_code == 200
+
+        ds = resp.json()
+        validate_json(instance=ds, schema=schema_objectlist)
+
+        # try to create one
+        suffix = random.choice(range(0, 1000))
+        rname = f'redhat.io.{suffix}'
+        payload = {
+            'name': rname,
+            'url': 'https://registry.redhat.io',
+        }
+        resp = uclient.post('_ui/v1/execution-environments/registries/', payload=payload)
+        assert resp.status_code == 201
+        rds = resp.json()
+        validate_json(instance=rds, schema=schema_ee_registry)
+
+        # try to get it by pulp_id
+        resp = uclient.get(f"_ui/v1/execution-environments/registries/{rds['pk']}/")
+        assert resp.status_code == 200
+        rds = resp.json()
+        validate_json(instance=rds, schema=schema_ee_registry)
+
+        # sync it
+        resp = uclient.post(
+            f"_ui/v1/execution-environments/registries/{rds['pk']}/sync/",
+            payload={}
+        )
+        assert resp.status_code == 202
+        task = resp.json()
+        validate_json(instance=task, schema=schema_task)
+
+        # wait for sync to finish
+        counter = 0
+        state = None
+        while state in [None, 'waiting', 'running']:
+            counter += 1
+            if counter >= 60:
+                raise Exception('ee registry sync task is taking too long')
+            resp = uclient.get(absolute_url=task['task'])
+            assert resp.status_code == 200
+            ds = resp.json()
+            validate_json(instance=ds, schema=schema_task_detail)
+            state = ds['state']
+            if state == 'completed':
+                break
+            time.sleep(.5)
```

Comment:
might be nice to use the constant we use for other polling, if appropriate https://github.com/ansible/galaxy_ng/blob/4499ab40e475d968b35068bb85d6ad30ff86c52d/galaxy_ng/tests/integration/constants.py#L8

Reply:
done